RFC-0010-jdk-upgrade.md

```diff
@@ -0,0 +1,293 @@
+As of today (2025-01-24) the version we use is JDK 8
+
+
+## Background
+
+The Java community has slowly but surely been moving to JDK versions beyond JDK 8  
+
+Some of our dependencies, e.g [Jetty](https://github.com/jetty/jetty.project/) have already moved to JDK 17, and declared prior versions to be out-of-support(See https://github.com/jetty/jetty.project/issues/10485. The latest supported version is Jetty 12)
+
+Not using modern versions of our dependencies opens Presto installations to security risks through CVEs, which hurts adoption of Presto
+
+See this [Prestocon 2024 presentation](https://www.youtube.com/watch?v=K6CjuPZtqW4&list=PLJVeO1NMmyqW_qoMMEyVq-wW8lBSxA7hP&index=10) for a deep-dive motivating the issue further (Slides [here TODO]())
+
+
+## Proposed Migration Plan  
+
+To migrate to newer versions of the JDK, we propose a phased approach -   
+
+![slide from Prestocon 2024 presentation](RFC-0010/jdk_upgrade_timeline.png)
+
+The salient points - 
+
+ - During the upgrade, the project will be built with two JDK versions , the current and the new target version (e.g JDK 8 and 17) simultaneously
+ - We will make use of our CI pipelines to run the full breadth of unit, integration and end-to-end product tests with every new JDK version
+ - Users will be invited to test new builds with their production workloads to report any issues observed around correctness, performance and stability
+ - As blocking issues shake out and get resolved, we will start deprecating support for older JDK versions
+
+ _Note : The timeline mentioned in this image is a proposal. Actual rollout timelines will differ_
+
+## Current Blockers
+We list the current blockers for the migration effort below
+
+### Presto On Spark
+[Presto on Spark](https://github.com/prestodb/presto/issues/13856)(PoS) integrates Presto's SQL engine with Apache Spark's distributed computing capabilities, allowing Presto queries to leverage Spark's scalability and fault tolerance for large-scale batch processing workloads. 
+
+While the Spark project has been putting out releases supporting newer JDK versions, Presto's users may not have migrated to the latest Spark version
+
+Here's is a summary of the max version that a Spark application can have, by Spark version (see [link](https://community.cloudera.com/t5/Community-Articles/Spark-and-Java-versions-Supportability-Matrix/ta-p/383669) for more details)
+
+
+| | |
+| --- | --- | 
+| **Spark Version** | **Supported Java Version(s)** |
+| 4.x (unreleased) | Java 8/11/17/21 | 
+| 3.3.0 - 3.5.1 | Java 8/11/17 | 
+| 3.0.0 - 3.2.4 | Java 8/11 | 
+| 2.4.0 - 2.4.8 | Java 8 |
+
+
+This means, for example, if we upgrade the bytecode level of Presto to that of JDK 17, users would only be able to use Presto On Spark on Spark 3.3.0+ clusters
+
+### Workaround
+The [presto-spark](https://github.com/prestodb/presto/tree/498784bb35feaf2787da865865ffe8ab3e8673ca/presto-spark) maven module has the below dependency graph -
+
+```
```

Comment:
For the rendered RFC, it might be better to put this in a ["details" block](https://docs.github.com/en/get-started/writing-on-github/working-with-advanced-formatting/organizing-information-with-collapsed-sections)